var/spack/repos/builtin/packages/extrae/package.py

```diff
@@ -35,54 +35,63 @@ class Extrae(Package):
        OpenMP, CUDA, OpenCL, pthread, OmpSs"""
     homepage = "https://tools.bsc.es/extrae"
     url      = "https://ftp.tools.bsc.es/extrae/extrae-3.4.1-src.tar.bz2"
+    version('3.7.1', sha256='95810b057f95e91bfc89813eb8bd320dfe40614fc8e98c63d95c5101c56dd213')
     version('3.4.1', '69001f5cfac46e445d61eeb567bc8844')
 
     depends_on("mpi")
-    depends_on("dyninst")
     depends_on("libunwind")
     depends_on("boost")
     depends_on("libdwarf")
     depends_on("papi")
     depends_on("elf", type="link")
     depends_on("libxml2")
-
+    depends_on("numactl")
+    depends_on("binutils+libiberty")
+    depends_on("gettext")
     # gettext dependency added to find -lintl
     # https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined
-    depends_on("gettext")
-    depends_on("binutils+libiberty")
 
-    def install(self, spec, prefix):
-        if 'openmpi' in spec:
-            mpi = spec['openmpi']
-        elif 'mpich' in spec:
-            mpi = spec['mpich']
-        elif 'mvapich2' in spec:
-            mpi = spec['mvapich2']
+    build_directory = 'spack-build'
 
-        extra_config_args = []
+    variant('dyninst', default=False, description="Use dyninst for dynamic code installation")
+    depends_on('dyninst@:9', when='+dyninst')
 
-        # This was added due to configure failure
-        # https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined
-        extra_config_args.append('LDFLAGS=-lintl')
+    variant('papi', default=True, description="Use PAPI to collect performance counters")
+    depends_on('papi', when='+papi')
+
+    def configure_args(self):
+        spec = self.spec
+        args = ["--with-mpi=%s" % spec['mpi'].prefix,
+                "--with-unwind=%s" % spec['libunwind'].prefix,
+                "--with-boost=%s" % spec['boost'].prefix,
+                "--with-dwarf=%s" % spec['libdwarf'].prefix,
+                "--with-elf=%s" % spec['elf'].prefix,
+                "--with-xml-prefix=%s" % spec['libxml2'].prefix,
+                "--with-binutils=%s" % spec['binutils'].prefix]
+
+        args += (["--with-papi=%s" % spec['papi'].prefix]
+                 if '+papi' in self.spec else
+                 ["--without-papi"])
+
+        args += (["--with-dyninst=%s" % spec['dyninst'].prefix]
+                 if '+dyninst' in self.spec else
+                 ["--without-dyninst"])
 
         if spec.satisfies("^dyninst@9.3.0:"):
             make.add_default_arg('CXXFLAGS=-std=c++11')
-            extra_config_args.append('CXXFLAGS=-std=c++11')
+            args.append('CXXFLAGS=-std=c++11')
```

Comment:
I would use `self.compiler.cxx11_flag` here so that it is compiler independent.

Reply:
Thanks, I didn't know that existed.